### PR TITLE
Task: Basics. Node.js fundamental theory

### PR DESCRIPTION
**Pull request by JieLiuRiver**
#### TASK 1.1

check `task1.js` file.

- [X] The program should be started from npmscript via nodemon
- [X] The program should be running in a stand-by mode and should not be terminated after the first-string processing.

#### TASK 1.2
check `task2.js` file.

- [X] Read the content of csvfile from `./csv` directory
- [X] Use the [csvtojsonpackage](https://github.com/Keyang/node-csvtojson) to convert csvfile to jsonobject.
- [x] Write the csvfile content to a new txtfile. Use the following format
- [x] Do not load all the content of the csvfile into RAM via stream (read/write file content line by line).
- [x] In case of read/write errors, log them in the console.
- [x] The program should be started via npm scriptusing nodemon

#### TASK 1.3
check `task3/` folder.

- [x] Rewrite the above-mentionedprograms to use babel(https://babeljs.io/)and ES6modules.

### CODE REVIEW

File: task2.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const csvToJson = csv()
+  .fromStream(readStream)
+  .pipe(writeStream)
+  .on('done', error => {
+    if (error) {
+      console.error(error);
+    } else {
+      console.log('Conversion completed successfully');
+    }
+  });
```
Comment (lczzlczz) [inline task2.js]: Hi @stupidWall, `csvToJson` here is an action and no returning from `csv()xxxx`.
I'd recommend to change it to:
```js
const csvToJson = () => {
  csv()
    .fromStream(readStream)
    .pipe(writeStream)
    .on('done', error => {
      if (error) {
        console.error(error);
      } else {
        console.log('Conversion completed successfully');
      }
    });
}

csvToJson();
```

Comment (lczzlczz) [reply task2.js]: Further, we can do this:
```js
const csvToJson = (sour, dest) => {
  const readableStream = fs.createReadStream(sour);
  const writableStream = fs.createWriteStream(dest);

  csv()
    .fromStream(readableStream)
    .pipe(writableStream)
    .on('done', error => {
      if (error) {
        console.error(error);
      } else {
        console.log('Conversion completed successfully');
      }
    });
}

csvToJson(csvfilepath, jsonFilePath);
```

File: task1.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+// Write a program which reads a string from the standard input stdin, reverses it and then writes it to the standard output stdout.
+process.stdin.on("data", function(data) {
+    data = data.toString().split('').reverse().join('')
```
Comment (lczzlczz) [inline task1.js]: We don't recommend to override the `data`, how about creating a new variable `const reversedStr = data.toString().split('').reverse().join('')` ?

File: task3/task2.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
+const csv = require('csvtojson/v2')
+const fs = require('fs')
```
Comment (lczzlczz) [inline task3/task2.js]: please use `import` instead

File: task3/index.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import './task1.js'
+import main from './task2'
+
+main()
```
Comment (lczzlczz) [inline task3/index.js]: This seems no need.

File: task3/task2.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            if (error) {
+                console.error(error);
+            } else {
+                console.log('Conversion completed successfully');
+            }
+        });
+    return csvToJson
+}
+
+export default main
```
Comment (lcssptz) [inline task3/task2.js:25]: seems no need.

Comment (lczzlczz) [inline task3/task2.js]: seems no need

File: task2.js (hunk 1 above)
Comment (JieLiuRiver) [reply task2.js]: Yes, it's better.

File: task1.js (hunk 2 above)
Comment (JieLiuRiver) [reply task1.js]: Yes, It is not a good habit to change the input parameter data, let me declare a variable

File: task3/task2.js (hunk 3 above)
Comment (JieLiuRiver) [reply task3/task2.js]: ok

**Review by lczzlczz — APPROVED** [review]
LGTM, thanks @stupidWall

**Pull request merged**